### DISCUSSION TITLE
Do I really need to use ORM in fastAPI?

### DISCUSSION THREAD

**Discussion by ide127** [fastapi/fastapi#6290] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import aiomysql

Class DataBase():

    async def __init__(self, loop):
        self.conn = await aiomysql.connect( ... )

    async def execute(self, query, loop):
        cur = await self.conn.cursor()
        sql_result = cur.execut(query)
        return sql_result
```

### Description

hello. I am a developer who developed with flask and is considering moving to fastAPI. When developing on flask, i use the above class to insert or retrieve data from DB. However, in fastAPI (perhaps only sqlAlchemy is supported), there seems to be no usages anywhere to using 'execute' method to get actual DB data rows.

I try to avoid using sqlAlchemy as much as possible. I don't like going through a layer of abstraction one more level to get data that exists in DB. Rather than get help from an ORM, I want to write and tune the actual SQL statements myself. Also, the DB I would to access already has all the tables and systems. This means that i don't need to specify the table structure like BaseModel in ORM.

So I want to find a way to access mysql directly from fastAPI to create and execute queries. 
I would be very grateful if you could help me or introduce a useful article.

Thank you vary much.

### Operating System

macOS

### FastAPI Version

0.79.1

### Python Version

3.9

**Comment by alexedtionweb** [comment, marked as answer]
I think you should use stackoverflow or discussions for such questions. GitHub issue is used to track bugs, feature requests, and potential improvements.

About you question, as the documentation state, you don't need a ORM to connect with the database, you can create you own connection using any driver you want. 

You can find a example at this repo:

https://github.com/s3rius/FastAPI-template

**Comment by ide127** [comment]
@alexedtionweb 
Thank you for the reply. From now on, the question will be posted on github.

I was encouraged by your words and decided to implement the db connection layer myself. 
I think i should probably create a database connection pool under the app = fastAPI() line.
And I think that the code will be executed only the first time when build, and I think that multiple connections existing in that unique and shared accessible connection pool can be recycled by each request.

If successful, I will contribute to a more robust fastAPI ecosystem by posting a review on my blog. thank you
